### ISSUE TITLE
[BUG]: Failing initial migration

### ISSUE THREAD

**Issue by ddelange** [getlago/lago-api#800]
**Describe the bug**
Hi 👋 

We would like to try out lago, but the startup sequence (initial migration) will crash.

**To Reproduce**
- Create a fresh postgres database
- Set DATABASE_URL
- Start up lago-api container

**Expected behavior**
Container runs through all migrations succesfully, and is ready to serve /graphql (so we can sign up and try it out).

**Screenshots**
[lago-api.log](https://github.com/getlago/lago-api/files/10660432/lago-api.log)

It looks like the app crashes on this migration: https://github.com/getlago/lago-api/blob/v0.21.0-beta/db/migrate/20220315141638_change_memberships_role_datatype.rb#L3-L4

It seems like it does not wait for the column to be dropped before trying to recreate the column. 

**Additional context**
- We're running on a cockroachlabs.cloud managed pg cluster
- RAILS_ENV=production

**Version**
- Lago: v0.21.0-beta

Any help is appreciated!

**Comment by ddelange** [comment]
hotfix:
```ruby
class ChangeMembershipsRoleDatatype < ActiveRecord::Migration[7.0]
  disable_ddl_transaction!
  def change
    execute "SET enable_experimental_alter_column_type_general = true"
    change_column :memberships, :role, :integer
    change_column_default :memberships, :role, default: 0
    change_column_null :memberships, :role, null: false
  end
end
```

**Comment by ddelange** [comment]
next migration failing: https://github.com/getlago/lago-api/blob/v0.21.0-beta/db/migrate/20220316115408_change_memberships_id_datatype.rb

```
-- change_table(:memberships)
rake aborted!

PG::InvalidColumnReference: ERROR:  column "id" is referenced by the primary key
```

**Comment by ddelange** [comment]
is there a chance these migrations can be squashed?

**Comment by jdenquin** [comment]
Hello @ddelange 

We are taking a look at this issue and come back to you asap.

**Comment by jdenquin** [comment]
We just tried to reproduce but we can't obtain the same error, can you explain how you are running the `api` on your side.

BTW we will squash some migrations on the next release!

**Comment by ddelange** [comment]
Hi @jdenquin, thanks for the quick reply! We're running on k8s, so just a simple Pod with start command `scripts/start.sh`, basically adopted your docker-compose file.

The DATABASE_URL points to cockroachdb, my next possible culprit as it's a pg flavor and not a straight up pg.

I looked at their [docs](https://www.cockroachlabs.com/docs/stable/postgresql-compatibility.html) and found nothing super relevant, apart from some [special constraint and approach](https://www.cockroachlabs.com/docs/v22.2/drop-constraint#drop-and-add-a-primary-key-constraint) around dropping and re-adding a primary key. This might be relevant for the last error I posted 🤔
